### ISSUE TITLE
cpu= -inf and gpu=-108.34196472167969

### ISSUE THREAD

**Issue by beanduan22** [pytorch/pytorch#180105]
### 🐛 Describe the bug

```
import torch
import torch.nn as nn

class Model(nn.Module):
    def __init__(self):
        super().__init__()
        self.fc1 = nn.Linear(8, 16)
        self.fc2 = nn.Linear(16, 4)

    def forward(self, x):
        y = self.fc1(x)
        v = torch.arccosh(y + 2)
        u = self.fc2(v)
        return torch.logdet(u), u.reciprocal_()

torch.manual_seed(42)
m_cpu = Model().eval()
torch.manual_seed(42)
m_gpu = Model().cuda().eval()
m_gpu.load_state_dict(m_cpu.state_dict())

x = torch.full((4, 8), 0.4033372700214386)

with torch.no_grad():
    cpu_log, cpu_u = m_cpu(x)
    gpu_log, gpu_u = [o.cpu() for o in m_gpu(x.cuda())]

print("CPU logdet:", cpu_log.item())
print("GPU logdet:", gpu_log.item())
print("CPU is inf:", torch.isinf(cpu_log).item(), "GPU is nan:", torch.isnan(gpu_log).item())
```

### Versions

CPU logdet: -inf
GPU logdet: -108.34196472167969
CPU is inf: True GPU is nan: False

cc @ptrblck @msaroufim @eqy @jerryzh168 @tinglvv @nWEIdia @jianyuh @nikitaved @mruberry @walterddr @xwang233 @Lezcano

**Comment by malfet** [comment]
Can you run collect env or at least describe what OS/GPU and PyTorch version you are trying to use, as I can not reproduce this on say Google Colab

**Comment by RAJTripathi3030** [comment]
Hi @beanduan22, I tried reproducing this issue on my machine and the results I've got are different from yours, see following:
`The code is same line by line`
`Output` after execution : 
```
PyTorch: 2.11.0+cu130
CPU logdet: -111.53960418701172
GPU logdet: -108.34196472167969
CPU is inf: False GPU is nan: False
```
My env information : 
* PyTorch 2.11.0+cu130, Python 3.14.3, Arch Linux, GTX 1650 Ti.
* CPU -inf not reproduced, but CPU/GPU discrepancy is still there (-111.53 vs -108.34).

**Comment by malfet** [comment]
Closing per https://github.com/pytorch/pytorch/wiki/Finding-Or-Reporting-Issues#reporting-accuracy-problems but do not hesitate to open a new one with better argument why this discrepancy is not something one should expect
